hooks/tk-multi-publish2/basic/publish_node.py

```diff
@@ -0,0 +1,132 @@
+        return {}
+
+    @property
+    def item_filters(self):
+        """
+        List of item types that this plugin is interested in.
+        Only items matching entries in this list will be presented to the
+        accept() method. Strings can contain glob patters such as *, for example
+        ["katana.*", "katana.session"]
+        """
+        return ["houdini.node.*"]
+
+    def accept(self, settings, item):
+        """
+        Method called by the publisher to determine if an item is of any
+        interest to this plugin. Only items matching the filters defined via the
+        item_filters property will be presented to this method.
+        A publish task will be generated for each item accepted here. Returns a
+        dictionary with the following booleans:
+
+        - accepted: Indicates if the plugin is interested in this value at
+          all. Required.
+        - enabled: If True, the plugin will be enabled in the UI, otherwise
+          it will be disabled. Optional, True by default.
+        - visible: If True, the plugin will be visible in the UI, otherwise
+          it will be hidden. Optional, True by default.
+        - checked: If True, the plugin will be checked in the UI, otherwise
+          it will be unchecked. Optional, True by default.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process.
+
+        :returns: dictionary with boolean keys accepted, visible, required and enabled.
+        """
+        # Naively accept all renders which have valid outputs
+        item.expanded = False
+        if not item._children:
+            return {"accepted": False}
+        return {"accepted": True, "checked": False}
+
+    def validate(self, settings, item):
+        """
+        Validates the given item to check that it is ok to publish.
+
+        Returns a boolean to indicate validity.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+
+        :returns: True if item is valid, False otherwise.
+        """
+        return True
+
+    def publish(self, settings, item):
+        """
+        Inherits the publish data from the parent item to pass to the child
+        items.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process.
+        """
+        scene_publish_data = item.parent.properties.get("sg_publish_data")
+        if scene_publish_data:
+            self.logger.debug("Getting sg_publish_data '{}' from parent".format(scene_publish_data))
```

Comment:
Place dictionary at the end in case it is huge, use logger style substitution

```suggestion
            self.logger.debug(
                "Getting sg_publish_data from parent '%s'", scene_publish_data,
            )
```